cpp/src/round/round.cu

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/detail/round.hpp>
+#include <cudf/round.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <type_traits>
+
+namespace cudf {
+namespace detail {
+namespace {  // anonymous
+
+float __device__ generic_round(float f) { return roundf(f); }
+double __device__ generic_round(double d) { return ::round(d); }
+
+float __device__ generic_modf(float a, float* b) { return modff(a, b); }
+double __device__ generic_modf(double a, double* b) { return modf(a, b); }
+
+template <typename T, typename std::enable_if_t<std::is_signed<T>::value>* = nullptr>
+T __device__ generic_abs(T value)
+{
+  return abs(value);
+}
+
+template <typename T, typename std::enable_if_t<not std::is_signed<T>::value>* = nullptr>
+T __device__ generic_abs(T value)
+{
+  return value;
+}
+
+template <typename T, typename std::enable_if_t<std::is_signed<T>::value>* = nullptr>
+bool __device__ is_negative(T value)
+{
+  return value < 0;
+}
+
+// this is needed to suppress warning: pointless comparison of unsigned integer with zero
+template <typename T, typename std::enable_if_t<not std::is_signed<T>::value>* = nullptr>
+bool __device__ is_negative(T value)
+{
+  return false;
+}
+}  // anonymous namespace
```

Comment:
`round_fn` can go in anon namespace too.

Reply:
Done.